wild/tests/integration_tests.rs

```diff
@@ -220,21 +221,19 @@ fn host_supports_clang_with_tls_desc() -> bool {
     static CLANG_SUPPORTS_TLS_DESC: OnceLock<bool> = OnceLock::new();
 
     *CLANG_SUPPORTS_TLS_DESC.get_or_init(|| {
-        let mut echo_child = Command::new("echo")
-            .arg("int main() { return 0; }")
-            .stdout(Stdio::piped())
-            .spawn()
-            .expect("Failed to start echo process");
-        echo_child.wait().expect("Failed to wait for echo");
-        let echo_out = echo_child.stdout.expect("Failed to open echo stdout");
-
-        let clang = Command::new("clang")
+        let mut clang = Command::new("clang")
             .args(["-mtls-dialect=gnu2", "-x", "c", "-", "-o/dev/null"])
-            .stdin(Stdio::from(echo_out))
-            .output()
+            .stdin(Stdio::piped())
+            .spawn()
             .expect("Failed to run clang");
-
-        clang.status.success()
+        let mut stdin = clang.stdin.take().expect("Failed to open stdin");
+        std::thread::spawn(move || {
```

Comment:
Is spawning a thread necessary? AFAIK, it shouldn't be.

Reply:
I followed the example here: https://doc.rust-lang.org/std/process/struct.Stdio.html#method.piped.

Reply:
Make sense, although in this case using up the entire 4K buffer shouldn't be a problem.

Reply:
I tried not using the separate thread:
```rust
let mut clang = Command::new("clang")
            .args(["-mtls-dialect=gnu2", "-x", "c", "-", "-o/dev/null"])
            .stdin(Stdio::piped())
            .spawn()
            .expect("Failed to run clang");
        let mut stdin = clang.stdin.take().expect("Failed to open stdin");
        stdin
            .write_all("int main() { return 0; }".as_bytes())
            .expect("Write of a source file failed");
        clang.wait().expect("Wait failed").success()
```

and the test-case execution is stuck :)

Reply:
It would have gotten stuck because clang was waiting for the end of the input stream. If you do `drop(stdin)`, it should work.

Reply:
Great, not it works :)